.github/workflows/build.yml

```diff
@@ -66,14 +66,70 @@ jobs:
       - run: yarn lint:copyright
       - run: yarn build --filter={./packages/*}...
 
-      - name: Host build output on gh pages
-        uses: JamesIves/github-pages-deploy-action@4.1.3
+  deploy-gh-pages:
+    name: Push static files to github pages
+    needs: install
+    runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node 16.X
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+
+      - name: Cache node_modules
+        uses: actions/cache@v3
+        id: cache-node-modules
+        with:
+          path: |
+            node_modules
+            packages/*/node_modules
+            apps/*/node_modules
+            playgrounds/*/node_modules
+          key: modules-${{ hashFiles('yarn.lock') }}
+
+      - name: Build html test pages
+        run: yarn build --filter=itwinui-css
+
+      - name: Build portal
+        run: yarn workspace portal build --base "/iTwinUI/${{ github.event.number || '' }}"
```

Comment:
the [base](https://docs.astro.build/en/reference/cli-reference/#--base) needs to be specified because we are not deploying to the root of the url (the "root" would be `itwin.github.io`)

so this basically means it will run either:
```
# 937 is the PR number
yarn workspace portal build --base "/iTwinUI/937"
```
or if running in main branch, then:
```
yarn workspace portal build --base "/iTwinUI/"
```